src/components/activity/ActivityRegistrator.tsx

```diff
@@ -0,0 +1,188 @@
+      <View style={{ padding: 10, flexDirection: 'column', height: '100%',  justifyContent: 'space-evenly'}}>
+
+        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+          <View style={{flex: 1, flexGrow: 1}}>
+            <Avatar.Icon icon={route.params.icon} size={iconSizes.avatar} />
+          </View>
+          <DatePicker date={date} setDate={setDate}  />
+        </View>
+
+        <View style={{ flexDirection: 'row' }}>
+          <TimePicker now={new Date()} defaultTimeOffset={60} steps={steps} fromTime={fromTime} setFromTime={setFromTime} 
+            toTime={toTime} setToTime={setToTime} />
+        </View>
+
+        <KeyboardAvoidingView>
+          <SuggestiveTextInput label={lang.activityRegistratorTextInputLabel} textInputText={activityText} setTextInputText={setActivityText} 
+            choises={ choises } choise={choise} setChoise={setChoise} />
+        </KeyboardAvoidingView>
+
+
+        <View style={{flexDirection: 'row', justifyContent: 'center'}}>
+
+          <View style={{flexDirection: 'column', width: '80%'}}>
+            
+            <View style={{ flexDirection: 'row'}}>
+              <Text>{lang.activityRegistratorImporanceLabel + ": "}</Text>
+              <Text>{importance}</Text>
+            </View>
+
+            <View style={{ flexDirection: 'row', alignItems: 'center'}}>
+              <Text>0</Text>
+              <Slider
+                style={{flex: 1}} value={5} step={1}
+                minimumValue={0} maximumValue={10}
+                onValueChange={(value: number) => {setImportance(value)}}
+                minimumTrackTintColor={colors.accent} maximumTrackTintColor="#000000"
+              />
+              <Text>10</Text>
+            </View>
+
+          </View>
+
+        </View>
+
+        <View style={{flexDirection: 'row', justifyContent: 'center'}}>
+
+          <View style={{flexDirection: 'column', width: '80%'}}>
+
+            <View style={{ flexDirection: 'row'}}>
+              <Text>{lang.activityRegistratorEnjoymentLabel + ": "}</Text>
+              <Text>{enjoyment}</Text>
+            </View>
+
+            <View style={{ flexDirection: 'row', alignItems: 'center'}}>
+              <Text>0</Text>
+              <Slider style={{flex: 1}} value={5} step={1}
+                minimumValue={0} maximumValue={10}
+                onValueChange={(value: number) => {setEnjoyment(value)}}
+                minimumTrackTintColor={colors.accent} maximumTrackTintColor="#000000"
+              />
+              <Text>10</Text>
+            </View>
+
+          </View>
+
+        </View>
+
+
+        <View style={{ flexDirection: 'row', justifyContent: 'space-around'}}>
+          <IconButton icon='close' size={iconSizes.large} onPress={() => onCancel()} color={colors.cancel} />
+          <IconButton icon='check' size={iconSizes.large} onPress={() => onConfirm()} color={colors.confirm} />
+        </View>
+      </View>
+    </TouchableWithoutFeedback>
+  );
+};
```

Comment:
I can follow it, but perhaps it should be broken into smaller parts?

Reply:
We've tried this but without making every component take in all hooks they need through props there is not practical way of doing this. Simply making a render function within the primary function does not work as discussed here:
* https://stackoverflow.com/questions/61280400/defining-a-sub-render-function-in-render-vs-a-react-component
* https://stackoverflow.com/questions/46777832/react-noob-onchange-element-loses-focus/46791156#46791156